### PR TITLE
More Pathways Rules

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: app/models/rules/majority_unsheltered.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/boston-cas/blob/production/LICENSE.md
+###
+
+class Rules::MajorityUnsheltered < Rule
+  def clients_that_fit(scope, requirement, opportunity)
+    if Client.column_names.include?(:majority_sheltered.to_s)
+      scope.where(majority_sheltered: ! requirement.positive)
```
Comment (eanders) [inline app/models/rules/majority_unsheltered.rb:10]: Does this not need to ignore nil?

**Review by eanders — APPROVED** [review]

File: app/models/rules/enrolled_in_hmis_project_type.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    type_clauses = if requirement.positive
+      clauses.join(' or ')
+    else
+      clauses.join(' and ')
+    end
+
+    scope.where(type_clauses)
+  end
+
+  private def clause(column_name, positive)
```
Comment (eanders) [inline app/models/rules/enrolled_in_hmis_project_type.rb]: Are you actually using this?

File: app/models/rules/enrolled_in_hmis_project_type.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      a_t[type_columns[project_type.to_sym]].eq(requirement.positive).to_sql
+    end.compact
+
+    type_clauses = if requirement.positive
+      clauses.join(' or ')
+    else
+      clauses.join(' and ')
+    end
+
+    scope.where(type_clauses)
```
Comment (eanders) [inline app/models/rules/enrolled_in_hmis_project_type.rb]: Does this need an `Arel.sql(type_clauses)` Looks like it's just passing plain sql to where.

File: app/models/rules/majority_unsheltered.rb (hunk 1 above)
Comment (awisema) [reply app/models/rules/majority_unsheltered.rb:10]: I don't think so -- SQL NULL is neither true nor false.

File: app/models/rules/enrolled_in_hmis_project_type.rb (hunk 2 above)
Comment (awisema) [reply app/models/rules/enrolled_in_hmis_project_type.rb]: Nope! Removed.

File: app/models/rules/enrolled_in_hmis_project_type.rb (hunk 3 above)
Comment (awisema) [reply app/models/rules/enrolled_in_hmis_project_type.rb]: Fixed.

**Review by eanders — APPROVED** [review]

**Pull request merged**